dags/legacy_test/tests/pytorch/r2.3/common.libsonnet

```diff
@@ -0,0 +1,215 @@
+    tpuSettings+: {
+      softwareVersion: 'tpu-ubuntu2204-base',
+    },
+    imageTag: 'r2.3.0-rc2_3.10',
+  },
+  PyTorchTest:: common.PyTorchTest + r2_3 {
+    local config = self,
+
+    podTemplate+:: {
+      spec+: {
+        initContainerMap+:: {
+          'tpu-version': {
+            image: config.podTemplate.spec.containerMap.train.image,
+            env+: [
+              {
+                name: 'TPU_NAME',
+                valueFrom: {
+                  fieldRef: {
+                    fieldPath: "metadata.annotations['name.cloud-tpus.google.com/train']",
+                  },
+                },
+              },
+            ],
+            command: [
+              'python3',
+              '-c',
+              |||
+                import importlib_metadata
+                import os
+                import re
+
+                import cloud_tpu_client
+
+                requirements = importlib_metadata.requires('torch_xla')
+                libtpu_pattern = r'libtpu-nightly ?@ https:\/\/storage.googleapis.com\/cloud-tpu-tpuvm-artifacts\/wheels\/libtpu-nightly\/libtpu_nightly-\d.\d.dev(\d{8})-\w+-\w+-\w+.whl'
+                libtpu_matches = [
+                  re.findall(libtpu_pattern, req)[0]
+                  for req in requirements
+                  if re.match(libtpu_pattern, req)
+                ]
+                assert len(libtpu_matches) == 1, f'{len(libtpu_matches)} matches in {requirements} (pattern: `{libtpu_pattern}`)'
+                libtpu_date = libtpu_matches[0]
+                print('libtpu date:', libtpu_date)
+
+                ctc = cloud_tpu_client.Client(tpu=os.path.basename('$(TPU_NAME)'), zone=os.path.dirname('$(TPU_NAME)'))
+                ctc.wait_for_healthy()
+                ctc.configure_tpu_version(f'pytorch-2.3-dev{libtpu_date}', restart_type='always')
+                ctc.wait_for_healthy()
+              |||,
+            ],
+          },
+        },
+      },
+    },
+  },
+  Functional:: mixins.Functional {
+    schedule: '0 6 * * *',
+    tpuSettings+: {
+      preemptible: false,
+    },
+  },
+  Convergence:: mixins.Convergence,
+  PyTorchTpuVmMixin:: experimental.PyTorchTpuVmMixin + experimental.PjRt {
+    local config = self,
+
+    tpuSettings+: {
+      softwareVersion: 'tpu-ubuntu2204-base',
+      tpuVmPytorchSetup: |||
+        pip3 install -U setuptools
+        # `unattended-upgr` blocks us from installing apt dependencies
+        sudo systemctl stop unattended-upgrades
+        sudo apt-get -y update
+        sudo apt install -y libopenblas-base
+        # for huggingface tests
+        sudo apt install -y libsndfile-dev
+        # Install torchvision by pinned commit in PyTorch 2.3 release branch.
+        pip install https://storage.googleapis.com/pytorch-xla-releases/wheels/tpuvm/torch-2.3.0rc2-cp310-cp310-linux_x86_64.whl
+        pip install --user --no-use-pep517 "git+https://github.com/pytorch/vision.git@2c127da8b5e2e8f44b50994c6cb931bcca267cfe"
```

Comment:
For torchvision and torch, can you install the upstream RCs instead? There will be instructions in the release announcement.

Reply:
There is no upstream RC whl released yet. I tried `pip install torch==2.3.0`

Reply:
PyTorch releases RCs through their private index. There are examples commands here: https://dev-discuss.pytorch.org/t/pytorch-2-3-release-branch-cut-for-pytorch-core-is-completed/1939

It looks like PyTorch is there, but not `torchvision` (yet)

Reply:
Sounds good, updated to `pip install torch==2.3.0 https://download.pytorch.org/whl/test/cpu`